### PR TITLE
Align NewExtensionPage with CreateCodeMonitorPage

### PR DESCRIPTION
**Pull request by eseliger**
Rob and I talked about how unaligned our create forms are, so I took a stab here at making the `NewExtensionPage` align with the design of the `CreateCodeMonitorPage`. In a separate PR, I will align the `CreateBatchChangePage` with this general layout as well.

| **Old** | **New** |
| --- | --- |
| image | image |

## Test plan

Tested manually in the browser, this is just some visual changes and functionality seems to be preserved.

### CODE REVIEW

File: client/web/src/enterprise/extensions/registry/backend.ts [hunk 1]
```diff
@@ -67,3 +67,37 @@ export function queryViewerRegistryPublishers(): Observable<GQL.RegistryPublishe
         })
     )
 }
+
+export function createExtension(
```
Comment (eseliger) [inline client/web/src/enterprise/extensions/registry/backend.ts:71]: just moved over, to get less expensive recompiles while working on the page.

File: client/web/src/extensions/extension/ExtensionAreaHeader.tsx [hunk 2]
```diff
@@ -209,9 +209,11 @@ export const ExtensionAreaHeader: React.FunctionComponent<ExtensionAreaHeaderPro
                                                     activeClassName="active"
                                                     exact={exact}
                                                 >
-                                                    {Icon && <Icon className="icon-inline" />}{' '}
-                                                    <span className="text-content" data-tab-content={label}>
-                                                        {label}
+                                                    <span>
```
Comment (eseliger) [inline client/web/src/extensions/extension/ExtensionAreaHeader.tsx:212]: Fixes nav item layout.

**Comment by rrhyne** [comment]
Nice work finding a design-debt issue and correcting it!

**Review by rrhyne — APPROVED** [review]
Looks good to me.

File: client/web/src/enterprise/extensions/registry/RegistryNewExtensionPage.tsx [hunk 3]
```diff
@@ -204,28 +169,28 @@ export const RegistryNewExtensionPage = withAuthenticatedUser(
...
+                                isErrorLike(this.state.publishersOrError) ||
+                                this.state.publishersOrError === 'loading' ||
+                                this.state.creationOrError === 'loading'
+                            }
+                            variant="primary"
+                        >
+                            {this.state.creationOrError === 'loading' ? (
+                                <LoadingSpinner />
+                            ) : (
+                                <AddIcon className="icon-inline" />
```
Comment (courier-new) [inline client/web/src/enterprise/extensions/registry/RegistryNewExtensionPage.tsx:188]: Just one nit, it seems like on the code monitoring, code insights, and batch changes "Create" pages, we omit the "+" icon from the button here!

Might also be worth including a "Cancel" button to be even more consistent with the others?

Comment (eseliger) [reply client/web/src/enterprise/extensions/registry/RegistryNewExtensionPage.tsx:188]: Good finds, will do both!

**Pull request merged**